src/saltext/vmware/states/vmc_org_users.py

```diff
@@ -0,0 +1,417 @@
+            result=True,
+        )
+    else:
+        return vmc_state._create_state_response(
+            name=name, comment="User {} is already present".format(username), result=True
+        )
+
+
+def absent(
+    name,
+    hostname,
+    refresh_key,
+    org_id,
+    notify_users=False,
+    verify_ssl=True,
+    cert=None,
+):
+    """
+    Ensure a given user does not exist for the given organization.
+
+     CLI Example:
+
+      .. code-block:: bash
+
+           salt-call --local -c /Users/Documents/salt_Configuration/local/etc/salt/ state.single vmc_org_users.absent name="test@vmware.com" hostname=console-stg.cloud.vmware.com org_id=10e1092f-51d0-473a-80f8-137652fd0c39 refresh_key=refresh_key_value verify_ssl=False
+
+    name
+        Indicates the email Id identifying the user to be removed from the organization.
+
+    hostname
+        The host name of CSP.
+
+    refresh_key
+        API Token of the user which is used to get the Access Token required for VMC operations.
+
+    org_id
+        The ID of organization from which the users should be removed.
+
+    notify_users
+        (Optional) Indicates whether the users need to notify through email. Default value is true.
+
+    verify_ssl
+        (Optional) Option to enable/disable SSL verification. Enabled by default.
+        If set to False, the certificate validation is skipped.
+
+    cert
+        (Optional) Path to the SSL client certificate file to connect to VMC.
+        The certificate can be retrieved from browser.
+    """
+
+    # search for the users if it's found remove them from the org else say there is no users
+
+    username = name
+    log.info("Checking if User with username %s is present", username)
+    org_users_list = __salt__["vmc_org_users.list"](
+        hostname=hostname,
+        refresh_key=refresh_key,
+        org_id=org_id,
+        verify_ssl=verify_ssl,
+        cert=cert,
+    )
+
+    if "error" in org_users_list:
+        return vmc_state._create_state_response(
+            name=name,
+            comment="Failed to get users for given org : {}".format(org_users_list["error"]),
+            result=False,
+        )
+
+    org_user = None
+    for org_user in org_users_list["results"]:
+        if username == org_user["user"].get("username"):
+            user_id = org_user["user"].get("userId")
+            log.info("user found with username %s", username)
+            break
+        else:
+            org_user = None
+
+    if __opts__.get("test"):
+        log.info("vmc_org_users.absent is called with test option")
+        if org_user:
+            return vmc_state._create_state_response(
+                name=name,
+                comment="State absent will remove user with username {}".format(username),
+            )
+        else:
+            return vmc_state._create_state_response(
+                name=name,
+                comment="State absent will do nothing as no user found with username {}".format(
```

Comment:
**suggestion** the `test` mode should only be different when salt *would* have changed anything. This should be rewritten:

```
if org_user:
    if __opts__.get("test"):
        # from line 374
        ...
    else:
        # from line 387
        removed_org_user = ...
        ...
else:
    # from line 409
    log.info("No user found ...")
    return ...
```

Reply:
If we test with valid email then all the use cases will be covered.
When we are testing with test account (for ex. "test@vmware.com"), it will always return "State absent will do nothing as no user found with username..." in the "test mode" because this test account is not part of vmware active directory and hence it would not be get added to the org while running "invite state" on this account.

Reply:
The problem isn't covering all of the cases it's that some of the messaging is **wrong**.

Cases:
 
- test mode
   - user found (A)
   - no user found (B)
- normal/not test mode
   - user found (C)
   - no user found (D)
   
But that's slightly incorrect because B and D _are the same case_. If there is no user found then it doesn't matter. We won't delete the user, nor would we really delete the user.

To see an example:

    salt-call --local state.single user.absent fizzygig test=True

Try this both with `test=True` and without.  There is zero difference in the message. We should do the same thing here:

```
if org_user:
    if __opts__.get('test'):
        # would have removed user (A)
    else:
        # actually removed user (C)
else:        
    # no user found (B & D)
```

Reply:
Updated